scripts/zones/Misareaux_Coast/mobs/Bloody_Coffin.lua

```diff
@@ -0,0 +1,40 @@
+-----------------------------------
+-- Area: Misareaux Coast
+-- Mob: Bloody Coffin
+-- Quest NM "Call of the Sea"
+-----------------------------------
+---@type TMobEntity
+local entity = {}
+
+local function usingAbility(mob)
+    local action = mob:getCurrentAction()
+    if
+        action == xi.action.MAGIC_CASTING or
+        action == xi.action.MAGIC_FINISH or
+        action == xi.action.MOBABILITY_START or
+        action == xi.action.MOBABILITY_USING
+    then
+        return true
+    end
+
+    return false
+end
```

Comment:
Reminder for myself: This logic could live in the behavior global if similar cases exist